Data/GraphViz/Types/Graph.hs

```diff
@@ -271,13 +271,17 @@ addNode :: (Ord n)
            -> Attributes
            -> DotGraph n
            -> DotGraph n
-addNode n mc as dg
-  | n `M.member` ns = error "Node already exists in the graph"
-  | otherwise       = addEmptyCluster mc
-                      $ dg { values   = ns' }
+addNode n mc as dg = addEmptyCluster mc $ dg { values = ns' }
   where
     ns = values dg
-    ns' = M.insert n (NI mc as M.empty M.empty) ns
+    ns' = M.insertWith mergeLogic n (NI mc as M.empty M.empty) ns
+    mergeLogic (NI newClust newAttrs newPreds newSuccs) (NI oldClust oldAttrs oldPreds oldSuccs) =
+        NI resClust resAttrs resPreds resSuccs
+      where
+        resClust = newClust <|> oldClust
+        resAttrs = S.toList $ S.union (S.fromList newAttrs) (S.fromList oldAttrs)
```

Comment:
This is probably wrong.  We don't want to rely upon `==` on Attributes as it means that e.g. two labels will be allowed through.

You want to use `Data.GraphViz.Attributes.Same` (and double-check the ordering of `mergeLogic` `S.union` to make sure that the new attributes have precedence).